src/pages/Transfer/CrossChainTransferForm/index.tsx

```diff
@@ -1,52 +1,187 @@
 
+import * as React from 'react';
 import { useTranslation } from 'react-i18next';
 import { withErrorBoundary } from 'react-error-boundary';
-import { useSelector } from 'react-redux';
+import {
+  useSelector,
+  useDispatch
+} from 'react-redux';
+import { useForm } from 'react-hook-form';
+import { toast } from 'react-toastify';
+import { InjectedAccountWithMeta } from '@polkadot/extension-inject/types';
 
-import AvailableBalanceUI from 'components/AvailableBalanceUI';
 import Accounts from 'components/Accounts';
 import Chains from 'components/Chains';
 import TokenField from 'components/TokenField';
 import ErrorFallback from 'components/ErrorFallback';
 import FormTitle from 'components/FormTitle';
 import SubmitButton from 'components/SubmitButton';
-import { COLLATERAL_TOKEN_SYMBOL } from 'config/relay-chains';
-import { displayMonetaryAmount } from 'common/utils/utils';
-import { StoreType } from 'common/types/util.types';
+import ErrorModal from 'components/ErrorModal';
+import {
+  COLLATERAL_TOKEN,
+  COLLATERAL_TOKEN_SYMBOL
+} from 'config/relay-chains';
+import { showAccountModalAction } from 'common/actions/general.actions';
+import { getUsdAmount } from 'common/utils/utils';
+import {
+  StoreType,
+  ParachainStatus
+} from 'common/types/util.types';
+
+import { ChainType } from 'common/types/chains.types';
+import STATUSES from 'utils/constants/statuses';
+import { xcmTransfer } from 'utils/relayChainApi/transfer';
+import { newMonetaryAmount } from '@interlay/interbtc-api';
+
+const TRANSFER_AMOUNT = 'transfer-amount';
+
+type CrossChainTransferFormData = {
+  [TRANSFER_AMOUNT]: string;
+}
 
 const CrossChainTransferForm = (): JSX.Element => {
+  const [destination, setDestination] = React.useState<InjectedAccountWithMeta | undefined>(undefined);
+  const [submitStatus, setSubmitStatus] = React.useState(STATUSES.IDLE);
+  const [submitError, setSubmitError] = React.useState<Error | null>(null);
+  const [approxUsdValue, setApproxUsdValue] = React.useState<string>('0');
+
+  const dispatch = useDispatch();
   const { t } = useTranslation();
 
   const {
-    collateralTokenTransferableBalance
+    register,
+    handleSubmit,
+    formState: { errors },
+    reset
+  } = useForm<CrossChainTransferFormData>({
+    mode: 'onChange'
+  });
+
+  const {
+    parachainStatus,
+    address,
+    prices
   } = useSelector((state: StoreType) => state.general);
 
+  const onSubmit = async (data: CrossChainTransferFormData) => {
+    if (!address) return;
+    if (!destination) return;
+
+    try {
+      setSubmitStatus(STATUSES.PENDING);
+
+      await xcmTransfer(
+        address,
+        destination.address,
+        newMonetaryAmount(data[TRANSFER_AMOUNT], COLLATERAL_TOKEN, true)
+      );
+
+      setSubmitStatus(STATUSES.RESOLVED);
+    } catch (error) {
+      setSubmitStatus(STATUSES.REJECTED);
+      setSubmitError(error);
+    }
```

Comment:
@tomjeatt 
I think we should use react-query's mutation mechanism which is good for DX and we already have examples in a good place in our codebase. In that case, we don't have to use `setSubmitError` and `setSubmitStatus` and as a result, the performance can be improved because one of the main factors which have a negative effect is unnecessary rendering.

Reply:
Understood, but out of scope at this point - this form is following the same pattern as the existing transfer form. I would like to improve this so we have consistency across the entire application, but for now I'm happy for the transfer and xcm transfer forms to be consistent with one another.

Reply:
OK, that sounds good for now. I think I will create a ticket.